### ISSUE TITLE
how to use this project

### ISSUE THREAD

**Issue by gaurav2695** [niyasc/Personality-Prediction-using-facebook-profile#1]
hi nyasc!! I want to know how to start with your project.... i configured already that which file is used for which part of the project but still don't able to get it how to execute it so that i can test it.

**Comment by niyasc** [comment]
This is basically a facebook application. 
- On loading application integrated with facebook, it will collect relevant user information and save it as `.data` files in `data` directory. Also, it will save user_id, user_name and access token in db.
- Then user will be redirected to another page, where user can choose between two options, which will show the predicted personality value and option to attend personality quiz.
- If the user has already attended the quiz they will be given with both results.

**Comment by gaurav2695** [comment]
thanks for this part and now will you please elaborate the database part, means which database you use and which web hosting server for making the app online.

**Comment by niyasc** [comment]
I was hosting the app on a free server, which is not available now. You may find a reference to the hosting provider here http://archive.is/xa19o

**Comment by gaurav2695** [comment]
please tell me more about database tables and schema which you used or about the tables attributes

**Comment by niyasc** [comment]
I don't have exact database details. Following is a rough structure:

Table : user_basic
Attributes : user_id, uname, access_token

Table : featurs
Attributes : extraversion, agreeableness, conscientiousness, neuroticism, openness

Table : questions
Attributes : sn, question, priority(0 if Disagree strongly stands for 5, 1 otherwise)

**Comment by gaurav2695** [comment]
niyasc, actually i most important thing i didn't get is how you calculated the percentage when asking questionnaire  to a person lets say he mark 0 to 5 according to his choice the how you get the percentage for 5 traits and compare with the neural network approach.

**Comment by niyasc** [comment]
I don't calculate any percentage of traits. It is percentage of error.

Error percentage = (actual value - predicted value) / 5 \* 100

**Comment by gaurav2695** [comment]
your app shows a set of questionnaire to the user and then user submit his/her responses to them. Then what will happen with these responses in this project ?

**Comment by niyasc** [comment]
See the process happening in this page
https://github.com/niyasc/Personality-Prediction-using-facebook-profile/blob/master/src/web_app/html/questionnire.php
- At the beginning, an array of 5 elements is initialized to - 0
- Then in each turn, result is added to respective arrays.
- In final turn, these results are divided with number of questions corresponding to each feature.
- Then finally, redirect user to home page, where a comparison between predicted personality and actual personality is shown.

**Comment by gaurav2695** [comment]
please tell me how the project specifically which files starts in sync when we start to play the app

**Comment by gaurav2695** [comment]
sir is https://github.com/niyasc/Personality-Prediction-using-facebook-profile/blob/master/src/web_app/ contains the final web application files then where i am unable to find where your had changed your appId and and appsecret in this project

**Comment by niyasc** [comment]
AppID and AppSecret are mentioned in this file> https://github.com/niyasc/Personality-Prediction-using-facebook-profile/blob/master/src/web_app/includes/initialize.php

**Comment by gaurav2695** [comment]
when i execute your app it shows error about this please help
![image](https://cloud.githubusercontent.com/assets/8272599/20551165/f174b9d4-b162-11e6-8340-4cdd9163b054.png)

**Comment by gaurav2695** [comment]
how you get permission in your app for the details from facebook profile because now when i try to build my app they also ask about "privacy policy url" i don't know about any of this please help !!

**Comment by gaurav2695** [comment]
please help me how will i create privacy policy for this app and how to integrate with this app

**Comment by gaurav2695** [comment]
Fatal error: SQLSTATE[HY000] [1045] ProxySQL Error: Access denied for user 'root'@'2a02:4780:bad:f00d::4' (using password: YES) in /home/004/231004/public_html/includes/functions.php on line 79 ??

I could not understand the error

**Comment by niyasc** [comment]
Problem is with your database connectivity. Please check your credentials. Also, facebook developer API changes over the time. So, please consult facebook developer guide or someone working on facebook app for more guidelines.

**Comment by gaurav2695** [comment]
finally how you use the pybrain library and how you do the k-fold cross
validation ?

**Comment by gaurav2695** [comment]
hi niyasc..

echo "<input type='hidden' name='feature".$question["sn"]."'
value=".$question["feature"].">";

this line of code is in src/webapp/templates/question.php at line 46

actually this line is getting things from 3rd database table of "question"
, but as you earlier replied in one of the previous mail that the 3rd table
has only 3 colums of serial number, question, and priority but in this line
of code i thought there was also an another attribute called "feature" in
that table. Please let me know whether i am correct or not if yes please
tell me what is it for..
thank you

**Comment by niyasc** [comment]
That is true. There might be a 3rd column to indicate the feature with which question is associated.

**Comment by gaurav2695** [comment]
and what about priority column in this question table ?? :/

and one more help nyasc please tell me how to convert this your query of
retrieving data from facebook to upgraded sdk version as you have used
earlier version and now i'am using latest version...

$facebook->get('/me?fields=id,name,gender,likes.limit(5000),
groups.limit(5000),friends.limit(5000).fields(gender), posts.limit(500)',
$access_token);

this gives me error that there is no groups and posts in my retrieved data
plz help

**Comment by gaurav2695** [comment]
also please tell me what this error is :


*Notice*: unserialize(): Error at offset 43566 of 45482 bytes in
*C:\xampp\htdocs\gtest\PersonalityPredictor.php* on line *14*

*i am getting when using your project .data files .. have you ever got this
typer of error  ??*
*plz help nyasC*

**Comment by gaurav2695** [comment]
plz tell me how you constructed the neural network in your project and
where to find database for training the database

**Comment by Prashanttomar1** [comment]
hi nyasc, can you tell me how to solve these problems?
![1st](https://user-images.githubusercontent.com/47498049/52728339-1bab4d80-2fdd-11e9-80a9-0d2ea111e566.png)
![2nd](https://user-images.githubusercontent.com/47498049/52728352-24038880-2fdd-11e9-8143-580d4e0d9a89.png)
